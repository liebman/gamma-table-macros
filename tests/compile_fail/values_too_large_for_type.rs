use gamma_table_macros::gamma_table;

// This should fail to compile because:
// - entry_type: u8 (max value 255)
// - size: 1024 
// - default max_value would be 1023 (size-1)
// - u8 cannot hold values larger than 255
gamma_table! {
    name: OVERFLOW_TABLE,
    entry_type: u8,
    gamma: 2.2,
    size: 1024
}

fn main() {} 