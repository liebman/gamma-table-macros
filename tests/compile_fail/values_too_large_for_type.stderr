error: max_value (1023) exceeds the maximum value (255) that can be stored in entry_type u8
 --> tests/compile_fail/values_too_large_for_type.rs:9:11
  |
9 |     name: OVERFLOW_TABLE,
  |           ^^^^^^^^^^^^^^
